core/vm/lfvm/hash_cache.go

```diff
@@ -0,0 +1,246 @@
+package lfvm
+
+import (
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"golang.org/x/crypto/sha3"
+)
+
+type hashCacheEntry32 struct {
```

Comment:
Perhaps you would like to add few lines that `key` is the input of the SHA3 function and `hash` is the output.

Reply:
Done